### DISCUSSION TITLE
BMS not count well SoC, alarm and correction ideas

### DISCUSSION THREAD

**Discussion by janekx** [Louisvdw/dbus-serialbattery#356] [Ideas]
Hi, i have meet the issue where battery was 55.2V and Daly BMS says 45%. It is issue of small carching and discharging currents where high current BMS (300A) not measure small currents precisly. Now in winter it become more often that charging currents are not hight like in summer. BMS count SoC on Ah charged and discharged and count from set battery capacity.
So i ask Daly to make self correct mechanism to correct SoC based on voltage about a once per 7 days or so. Waiting for theirs statement but second idea is to add to driver the correction. What we know is the voltage quite precisly and we know what voltage means what SoC. 48V system 54.4V and up is 100%, 52.8V is around 70%, 52V is around 40%. 
So the driver can check the voltage when is battery in standby and check SoC from BMS and if the diference is big can alarm user the provided info needs attention. As well can be done some correction to SoC from BMS for Victron but it needs some discussion and make procedures to work well. 
Is this idea interesting for you?

**Comment by transistorgit** [comment]
We just added an option to manually setting of SOC to the daly driver. I also would like to have an automatic reset like "if upper voltage limit reached, set et to 100%". I will start a discussion about that.
